src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -218,7 +218,9 @@ void Transformations::PreLpt(const std::vector<ov::element::Type>& defaultPrecis
             {ov::element::u16,     ov::element::i32},
             {ov::element::u32,     ov::element::i32},
             {ov::element::f64,     ov::element::f32},
+#if !defined(__ARM_FEATURE_FP16_VECTOR_ARITHMETIC)
```

Comment:
I would propose not to touch this conversion list for now. We don't now how transformation pipeline will be affected in case fp16 precision is kept in the model. That's needs to be properly validated.
So let the plugin convert fp16 model to canonical fp32 state, launch transofmation and them EnforcePrecision pass will do required mark-up fo fp16 based on plugin needs.